### ISSUE TITLE
[Accessibility]: Provide screen reader context for sublists of ref panel

### ISSUE THREAD

**Issue by varungandhi-src** [sourcegraph/sourcegraph#36606]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33510

### Problem description

When navigating through the ref panel, VoiceOver directly announces the repo names and file paths without saying that they are repo names or file paths. From the POV of a screen reader user, this seems a little disorienting, because there is a change in level of abstraction, but you only realize that after hearing a bit more.

### Expected behavior

It would be better for the screen reader to announce "repository" before it reads out the repository name and "file path" before it reads a file path.

### Additional details

[Example ref panel link](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/enterprise/internal/authz/gerrit/client.go?L9:6&subtree=true#tab=implementations_go). It can be triggered by hovering over an identifier and clicking on Find References or Find Implementations (if visible).

NOTE: There are two ref panels, the "new one" and the "old one"; the new one can be triggered by clicking the "Try the new reference panel experience!" Here is a screenshot of what the new ref panel looks like:

![reference panel](https://user-images.githubusercontent.com/93103176/171985257-843a7ab2-4041-4dd7-b19c-0997244e78b7.png)

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by GitStart. I have assigned the label 'gitstart' and added the assignee 'gitstart-sourcegraph'.
